Source/Public/New-BicepParameterFile.ps1

```diff
@@ -33,22 +36,29 @@ function New-BicepParameterFile {
     process {
         $File = Get-Item -Path $Path
         
-        if ($File) {
+        if ($AsHashTable.IsPresent) {
             $ARMTemplate = ParseBicep -Path $File.FullName -IgnoreDiagnostics
-
-            if($PSBoundParameters.ContainsKey('OutputDirectory')) {
-                $OutputFilePath = Join-Path -Path $OutputDirectory -ChildPath ('{0}.parameters.json' -f $File.BaseName)
-            }
-            else {
-                $OutputFilePath = $File.FullName -replace '\.bicep','.parameters.json'
-            }
-            if (!$PSBoundParameters.ContainsKey('Parameters')){
-                $Parameters='Required'
-            }
-             GenerateParameterFile -Content $ARMTemplate -Parameters $Parameters -DestinationPath $OutputFilePath -WhatIf:$WhatIfPreference
+            $ARMTemplateObject = ConvertFrom-Json -InputObject $ARMTemplate
+            $ARMTemplateObject | ConvertToHashtable -Ordered
         }
         else {
-            Write-Error "No bicep file named $Path was found!"
+            if ($File) {
+                $ARMTemplate = ParseBicep -Path $File.FullName -IgnoreDiagnostics
+                
+                if ($PSBoundParameters.ContainsKey('OutputDirectory')) {
+                    $OutputFilePath = Join-Path -Path $OutputDirectory -ChildPath ('{0}.parameters.json' -f $File.BaseName)
+                }
+                else {
+                    $OutputFilePath = $File.FullName -replace '\.bicep', '.parameters.json'
+                }
+                if (!$PSBoundParameters.ContainsKey('Parameters')) {
+                    $Parameters = 'Required'
+                }
+                GenerateParameterFile -Content $ARMTemplate -Parameters $Parameters -DestinationPath $OutputFilePath -WhatIf:$WhatIfPreference
```

Comment:
Do something like this and update `GenerateParameterFile` to support `-AsHashtable`
```powershell
                if ($AsHashTable.IsPresent) {
                    GenerateParameterFile -Content $ARMTemplate -Parameters $Parameters -AsHashtable -WhatIf:$WhatIfPreference
                }
                else {
                    GenerateParameterFile -Content $ARMTemplate -Parameters $Parameters -DestinationPath $OutputFilePath -WhatIf:$WhatIfPreference
                }
```